File: Fugazi/src/org/fugazi/qls/ast/segment/Page.java
package org.fugazi.qls.ast.segment;

import org.fugazi.qls.ast.IQLSASTVisitor;
import org.fugazi.qls.ast.style.DefaultStyleDeclaration;

import java.util.List;

public class Page extends Segment {

    public Page(int _lineNum, 
                String _name, 
                List<Section> _sections, 
                List<DefaultStyleDeclaration> _defaultStyles) 
    {
        super(_lineNum, _sections, _defaultStyles, _name);
    }

    public Page(String _name, 
                List<Section> _sections, 
                List<DefaultStyleDeclaration> _defaultStyles) 
    {
        super(_sections, _defaultStyles, _name);
    }

    public <T> T accept(IQLSASTVisitor<T> _visitor) {
        return _visitor.visitPage(this);
    }
}

File: CarlaErik/UvA.SoftCon.Questionnaire/QLS/AST/Model/Page.cs
using System.Collections.Generic;
using UvA.SoftCon.Questionnaire.Common.AST.Model;

namespace UvA.SoftCon.Questionnaire.QLS.AST.Model
{
    public class Page : QLSNode
    {
        public Identifier Id
        {
            get;
            private set;
        }

        public IEnumerable<Section> Sections
        {
            get;
            private set;
        }

        public IEnumerable<DefaultStyle> DefaultStyles
        {
            get;
            private set;
        }

        internal Page(Identifier id, IEnumerable<Section> sections, IEnumerable<DefaultStyle> defaultStyles, TextPosition position)
            : base(position)
        {
            Id = id;
            Sections = sections;
            DefaultStyles = defaultStyles;
        }

        public override T Accept<T>(IStyleSheetVisitor<T> visitor)
        {
            return visitor.VisitPage(this);
        }
    }
}
